src/artifactRegistry.ts

```diff
@@ -109,6 +198,13 @@ export class CachingDockerRegistryClient {
     },
   });
 
+  async getGitCommitsBetweenTags(
+    options: GitCommitsBetweenTagsOptions,
+  ): Promise<string[]> {
+    // For now we aren't caching anything since this will on run on promotion prs
```

Comment:
arguably the thing we want to cache here is the listTags call, not the higher level calculation.  but you're right that most of the time there's only one promotion being considered for any given docker image...  that said, it might make sense to make the DockerRegistryClient method be just the equivalent of the listTags which would allow you to mock that out and test the "find min" logic directly.